File: test-support/reference-data/scripts/r/moments.R
generate_moments <- function() {
  rows <- list()

  for (path in fixture_files("moments")) {
    case <- case_name(path)
    x <- read_fixture(path, "value", minimum_n = 4L)$value
    rows[[case]] <- checked_reference(data.frame(
      case = case,
      n = length(x),
      mean = base::mean(x),
      variance = stats::var(x),
      # Type 2 selects the adjusted sample skewness and excess kurtosis.
      skewness = e1071::skewness(x, type = 2),
      excess_kurtosis = e1071::kurtosis(x, type = 2)
    ), path)
  }

  write_reference(do.call(rbind, rows), "moments.csv")
}

File: crates/unrvl-stats/src/moments.rs
//! Moments and shape statistics.
//!
//! This module contains mean, sample variance, bias-corrected sample skewness,
//! and bias-corrected sample excess kurtosis.

mod extension;
mod kurtosis;
mod mean;
mod skewness;
mod var;

pub use extension::MomentsExt;
pub use kurtosis::excess_kurtosis;
pub use mean::mean;
pub use skewness::skewness;
pub use var::var;
